examples/llm/src/hf_causal_lm.py

```diff
@@ -42,20 +41,24 @@ def get_targets(self, batch: dict):
         targets[:, -1] = -100
         return targets
 
-    def forward(self, batch: dict):
+    def forward(self, batch: dict) -> CausalLMOutput:
         return self.model(input_ids=batch['input_ids'],
-                          attention_mask=batch['attention_mask'].bool()).logits
-
-    def eval_forward(self, batch: dict, outputs: Optional[Tensor] = None):
-        return outputs if outputs is not None else self.forward(batch)
```

Comment:
Be careful about switching to the parent `eval_forward` without changing some of the other methods in this class. I'm specifically worried about how the parent `eval_forward` pops `labels` out of the batch and stashes it in `self.labels`. This might require some re-writing of this class's `get_targets` and maybe `loss` too.

Reply:
Reverting this...